tests/osaka/eip7212_secp256r1_precompiles/test_secp256r1_precompile_module.py

```diff
@@ -0,0 +1,116 @@
+"""
+abstract: Tests [EIP-7212 EIP-7212: Precompiled for secp256r1 Curve Support](https://eips.ethereum.org/EIPS/eip-7212)
+    Test cases for [EIP-7212 EIP-7212: Precompiled for secp256r1 Curve Support](https://eips.ethereum.org/EIPS/eip-7212)].
+"""
+
+import pytest
+
+from ethereum_test_tools import (
+    Alloc,
+    Environment,
+    StateTestFiller,
+    Transaction,
+)
+from ethereum_test_tools import Opcodes as Op
+
+from .helpers import vectors_from_file
+from .spec import Spec, ref_spec_7212
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_7212.git_path
+REFERENCE_SPEC_VERSION = ref_spec_7212.version
+
+pytestmark = [
+    pytest.mark.valid_from("Osaka"),
+]
+
+
+@pytest.mark.parametrize(
+    "input_data,expected_output,vector_gas_value",
+    vectors_from_file("secp256r1_test.json"),
+)
+@pytest.mark.parametrize("precompile_address", [Spec.P256VERIFY], ids=[""])
+@pytest.mark.eip_checklist("new_precompile/test/call_contexts/normal")
+def test_valid(state_test: StateTestFiller, pre: Alloc, post: dict, tx: Transaction):
+    """Test P256Verify precompile."""
+    state_test(env=Environment(), pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.parametrize(
+    "input_data,expected_output,precompile_gas_modifier,call_succeeds",
+    [
+        pytest.param(
+            Spec.H0 + Spec.R0 + Spec.S0 + Spec.X0 + Spec.Y0,
+            Spec.SUCCESS_RETURN_VALUE,
+            1,
+            True,
+            id="extra_gas",
+        ),
+        pytest.param(
+            Spec.H0 + Spec.R0 + Spec.S0 + Spec.X0 + Spec.Y0,
+            Spec.INVALID_RETURN_VALUE,
+            -1,
+            False,
+            id="insufficient_gas",
+        ),
+    ],
+)
+@pytest.mark.parametrize("precompile_address", [Spec.P256VERIFY], ids=[""])
+@pytest.mark.eip_checklist("new_precompile/test/value_transfer/fee/over")
+@pytest.mark.eip_checklist("new_precompile/test/value_transfer/fee/under")
```

Comment:
Should be:
```suggestion
@pytest.mark.eip_checklist("new_precompile/test/gas_usage/constant/exact")
@pytest.mark.eip_checklist("new_precompile/test/gas_usage/constant/oog")
```

Reply:
I think we should add an explicit `exact` case here anyway just for the sake of completeness and readability, even though I understand it's implicitly checked in many other tests.

The `over` case we can leave the case but unmarked because it's not part of the checklist (we could add it to the checklist later).